### ISSUE TITLE
llvm 14 compatible release

### ISSUE THREAD

**Issue by 0-wiz-0** [ziglang/zig#12534]
### Zig Version

0.9.1

### Steps to Reproduce

Please make a release that works with llvm 14.
When building the latest release, 0.9.1, with llvm 14 installed, I get:
```
-- Detecting CXX compile features - done
Configuring zig version 0.9.1
CMake Error at cmake/Findllvm.cmake:99 (message):
  expected LLVM 13.x but found 14.0.6 using
  /tmp/lang/zig/work/.buildlink/bin/llvm-config
Call Stack (most recent call first):
  CMakeLists.txt:107 (find_package)
```

### Expected Behavior

Builds fine.

### Actual Behavior

Doesn't accept llvm 14.

**Comment by andrewrk** [comment]
Here is the current plan: 

Zig 0.10.0 will be released on September 21. There will not be any releases prior to this. If LLVM 15 has been released by then, then Zig 0.10.0 will depend on LLVM 15. Otherwise, Zig 0.10.0 will depend on LLVM 14.

**Comment by will-ca** [comment]
For reference, this appears to be the commit that began switching to LLVM 15:

https://github.com/ziglang/zig/commit/adb4a953021aadcfd539589eeccc2fe4e6ab827a

IIRC I was able to build the immediately preceding commit with LLVM+Clang 14.
